### ISSUE TITLE
EvalError: Refused to evaluate a string as JavaScript

### ISSUE THREAD

**Issue by chrisyip** [AtomLinter/linter-eslint#253]
```
EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".
  at Function (native)
  at Function.line.toFunction (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/node_modules/generate-function/index.js:55:21)
  at compile (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/index.js:543:23)
  at visit (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/index.js:326:16)
  at /Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/index.js:523:9
  at Array.forEach (native)
  at visit (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/index.js:520:31)
  at compile (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/index.js:537:3)
  at module.exports (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/node_modules/is-my-json-valid/index.js:564:10)
  at Object.<anonymous> (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/lib/testers/rule-tester.js:79:22)
  at Module._compile (module.js:428:26)
  at Object.defineProperty.value [as .js] (/Applications/Atom.app/Contents/Resources/app.asar/src/compile-cache.js:169:21)
  at Module.load (module.js:353:32)
  at Function.Module._load (module.js:308:12)
  at Module.require (module.js:363:17)
  at require (module.js:382:17)
  at Object.<anonymous> (/Users/Chris/.atom/packages/linter-eslint/node_modules/eslint/lib/api.js:11:17)
  at Module._compile (module.js:428:26)
  at Object.defineProperty.value [as .js] (/Applications/Atom.app/Contents/Resources/app.asar/src/compile-cache.js:169:21)
  at Module.load (module.js:353:32)
  at Function.Module._load (module.js:308:12)
  at Module.require (module.js:363:17)
  at require (module.js:382:17)
  at Object.module.exports.requireESLint (/Users/Chris/.atom/packages/linter-eslint/lib/linter-eslint.coffee:220:12)
  at Object.provider.lint (/Users/Chris/.atom/packages/linter-eslint/lib/linter-eslint.coffee:96:32)
  at promises.push.Promise.then._this.emitter.emit.linter (/Users/Chris/.atom/packages/linter/lib/linter-registry.js:95:28)
  at /Users/Chris/.atom/packages/linter/lib/linter-registry.js:94:25
  at Set.forEach (native)
  at LinterRegistry.lint (/Users/Chris/.atom/packages/linter/lib/linter-registry.js:92:20)
  at /Users/Chris/.atom/packages/linter/lib/linter.coffee:88:16
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
  at /Users/Chris/.atom/packages/linter/lib/editor-linter.js:100:23
  at /Users/Chris/.atom/packages/linter/lib/helpers.coffee:23:18
```

Atom 1.1.0 + latest Linter & Linter-ESLint.

**Comment by chrisyip** [comment]
If I create a `.eslintrc` in the project folder, everything goes well.

But I have checked `Disable When No Eslintrc File In Path`.

![image](https://cloud.githubusercontent.com/assets/189559/10877603/9f28ec92-8182-11e5-98c0-d29800c7928e.png)

**Comment by sushantdhiman** [comment]
I can confirm this as well. I am using `Ubuntu 14.04 Latest` with `Atom 1.1`

**Comment by lmartins** [comment]
Having the same issue.

**Comment by rumkin** [comment]
The same on MacOS.

**Comment by nelix** [comment]
Same issue here, I assume its because loophole needs to be configured to wrap eslints json validator?
I would have thought eslint already being wrapped in allowUnsafeNewFunction would have fixed this :/

**Comment by blackbeam** [comment]
Same. MacOs with Atom 1.1

**Comment by Tom910** [comment]
same problem,
Mac OS, atom 1.1, linter-eslint 4.0.0, global linter 1.8

**Comment by StefanoMagrassi** [comment]
Same problem on Debian 8.

Little suggest: when I set "disableWhenNoEslintrcFileInPath" to true and there isn't a .eslintrc file in project, the entire package should be "bypassed"

**Comment by AndrewHuffman** [comment]
Same problem here and for several on our development team.

@iam4x - You appear to be the main contributor, are you able to look at this?

**Comment by fastfrwrd** [comment]
Not sure how helpful it is, but here's the offending line, which is a downstream dependency of eslint itself it seems: https://github.com/mafintosh/generate-function/blob/master/index.js#L55

**Comment by puzzfuzz** [comment]
I ran into this as well. I was able to get around this by:
- installing `eslint` locally in the project with `npm install --save-dev eslint`
- adding a `.eslintrc` file to the project

**Comment by StefanoMagrassi** [comment]
@puzzfuzz yes, ok, if you do that everything works fine, but this is the problem: I **don't want** to install locally `eslint` (and/or add a `.eslintrc` file) in my project if I **don't need it** only in order to hide the error popup.

If my project doesn't support linting with eslint or linting at all, I want the package is disabled or at least silenced.

And this has to be "transparent", not explicitly set in some project configuration.

**Comment by lequeso** [comment]
Same problem here,

Ubuntu 14.04, Atom 1.1.0 + latest Linter & Linter-ESLint.

Using Global Eslint + setting Global Node Path  fixes it.

But, as @StefanoMagrassi  says, its should be transparent. e.g. when used together with nvm, setting Global Node Path every time isn't nice...

**Comment by rumkin** [comment]
It looks like the problem is not in linter-eslint itself in submodule of eslint called is-my-json-valid which uses module generate-function.

**Comment by marlowpayne** [comment]
Also getting this error on Mac OS, Atom v1.1.0, latest eslint installed globally (v1.9.0), latest linter and linter-eslint Atom packages.

Found that downgrading linter-eslint to v3.1.1 worked for me.

```bash
apm uninstall linter-eslint
apm install linter-eslint@3.1.1
```

**Comment by pitaj** [comment]
The `generate-function` module I believe is causing a permissions issue.

**Comment by wmertens** [comment]
Actually, it is calling `new Function` which is faster/safer (no scope included) but anyway :) The reason is that the javascript string it builds is tainted as coming from an unsafe source and needs to be made safe somehow. In Perl you do that by running it through a regex, not sure about js.

**Comment by nelix** [comment]
Seems like when not using global eslint or your projects local eslint, linter-eslint calls its own internal version of eslint but it doesnt call it via loophole.

I just put a PR up that should fix this

**Comment by steelbrain** [comment]
Fixed in v5.0.0
